packages/dds/tree/src/feature-libraries/modular-schema/fieldKind.ts

```diff
@@ -0,0 +1,161 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ChangeEncoder } from "../../change-family";
+import { ChangeRebaser } from "../../rebase";
+import {
+    FieldSchema, FieldKindIdentifier, TreeSchemaIdentifier, SchemaPolicy, fieldSchema, SchemaData,
+} from "../../schema-stored";
+import { isNeverField } from "./comparison";
+
+/**
+ * This logic lives in feature-libraries since using it is technically optional.
+ *
+ * This is all helper code for doing a dependency in version implementations of
+ * change rebasers, edit builders and such more modular.
+ */
+
+/**
+ * Functionality for FieldKinds that is stable,
+ * meaning that it can not change in any measurable way without providing a new identifier.
+ *
+ * It is assumed that this information and policy is available on all clients interacting with a document
+ * using the identifier.
+ *
+ * This must contain enough information to process remote edits to this FieldKind consistently with all clients.
+ * All behavior must be deterministic, and not change across versions of the app/library.
+ *
+ * These policies include the data encoding, change encoding, change rebase and change application.
+ */
+export class FieldKind {
+    /**
+     * @param identifier - Globally scoped identifier.
+     * @param multiplicity - bound on the number of children fields of this kind may have.
```

Comment:
This changes the meaning to something incorrect. This has nothing to do with a number of fields, it is about the number of children in a field of this kind.

Reply:
Ohhh, I see. Maybe this would make it easier to parse then?

```suggestion
     * @param multiplicity - bound on the number of children that fields of this kind may have.
```